src/screens/charging-stations/connector-details/ChargingStationConnectorDetails.tsx

```diff
@@ -727,27 +743,93 @@ export default class ChargingStationConnectorDetails extends BaseAutoRefreshScre
           <View style={style.noButtonStopTransaction} />
         )}
         {/* Details */}
-        <ScrollView
-          contentContainerStyle={style.scrollViewContainer}
-          refreshControl={<RefreshControl refreshing={this.state.refreshing} onRefresh={this.manualRefresh} />}>
-          <View style={style.rowContainer}>
-            {this.renderConnectorStatus(style)}
-            {this.renderUserInfo(style)}
-          </View>
-          <View style={style.rowContainer}>
-            {this.renderInstantPower(style)}
-            {this.renderTotalConsumption(style)}
-          </View>
-          <View style={style.rowContainer}>
-            {this.renderElapsedTime(style)}
-            {this.renderInactivity(style)}
-          </View>
-          <View style={style.rowContainer}>
-            {this.renderBatteryLevel(style)}
-            {isPricingActive ? this.renderPrice(style) : <View style={style.columnContainer} />}
-          </View>
-        </ScrollView>
+        {connector?.status === ChargePointStatus.AVAILABLE ? (
+          <ScrollView
+            contentContainerStyle={style.scrollViewContainer}
+            refreshControl={openUserModal ? null : <RefreshControl refreshing={this.state.refreshing} onRefresh={this.manualRefresh} />}>
+            <View style={style.rowContainer}>{this.renderConnectorStatus(style)}</View>
+            {isAdmin && this.renderUserSelection(style, formStyle)}
+          </ScrollView>
+        ) : (
+          <ScrollView
+            contentContainerStyle={style.scrollViewContainer}
+            refreshControl={<RefreshControl refreshing={this.state.refreshing} onRefresh={this.manualRefresh} />}>
+            <View style={style.rowContainer}>
+              {this.renderConnectorStatus(style)}
+              {this.renderUserInfo(style)}
+            </View>
+            <View style={style.rowContainer}>
+              {this.renderInstantPower(style)}
+              {this.renderTotalConsumption(style)}
+            </View>
+            <View style={style.rowContainer}>
+              {this.renderElapsedTime(style)}
+              {this.renderInactivity(style)}
+            </View>
+            <View style={style.rowContainer}>
+              {this.renderBatteryLevel(style)}
+              {isPricingActive ? this.renderPrice(style) : <View style={style.columnContainer} />}
+            </View>
+          </ScrollView>
+        )}
       </Container>
     );
   }
+
+  private openUserModal(open: boolean): void {
+    this.setState({ openUserModal: open });
+  }
+
+  private onUserSelected(users: User[]): void {
+    if (users && users.length > 0) {
```

Comment:
Use !Utils.isEmptyArray().
Handle userIDs now, no need the full user object I think for selection.

Reply:
I need to display the user's name